### PR TITLE
SRCH-2772 run CircleCi against multiple Elasticsearch versions

### PR DESCRIPTION
**Pull request by MothOnMars**
## Summary
This PR configures CircleCi to run our specs against both ES 6 and ES 7. It also:
- refactors the config to use CircleCi parameters instead of environmental variables for the versions. The env vars were a holdover from our old, home-grown test matrix, while parameters are the more canonical CircleCi approach: https://circleci.com/docs/2.0/reusing-config
- removes the ES 7 cluster from the CircleCi executor. The multiple clusters were required for a single spec, so I have simplified that spec by faking a second cluster.
- updates the test & dev environments to run ES 6 on the default port 9200, and Kibana on 5601.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review:

#### Functionality Checks

- [x] Code is functional. - I've tested a version of this branch without one of our deprecation fixes, and confirmed that it passes on ES 6, and fails on ES 7: https://app.circleci.com/pipelines/github/MothOnMars/search-gov?branch=test_es_6_7_2772_failing&filter=all

- [x] Automated checks pass, if applicable. If Code Climate checks do not pass, explain reason for failures:

- [x] If your changes will be tested manually, you have run `bundle update` and committed your changes to Gemfile.lock. - N/A
 
- [x] You have merged the latest changes from the target branch (usually `master` or `main`) into your branch.
 
- [x] If your target branch is NOT `master` or `main`, specify the reason:
 
- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release #.#.#** matching the release number
 
- [x] You have squashed your commits into a single commit (exception: your PR includes commits with formatting-only changes, such as required by Rubocop)

- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket
 
#### Process Checks

- [x] You have specified an "Assignee", and if necessary, additional reviewers

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -70,20 +65,23 @@ commands:
...
+    executor: 
+      name: test_executor
+      ruby_version: << parameters.ruby_version >>
+      elasticsearch_version: << parameters.elasticsearch_version >>
 
     steps:
       - checkout
 
       - run:
           name: Wait for Elasticsearch 6.8
```
Comment (lorawoodford) [inline .circleci/config.yml:83]: @MothOnMars Minor nit that can be ignored if you wish, but seeing this "Wait for Elasticsearch 6.8" step logged out in a ES7 run is a bit confusing.  Do we need this at all?  Does it need to be conditional on the elasticsearch_version parameter?  Everything runs/passes/fails as anticipated, of course, so more of a "this just looks weird" comment.

**Review by lorawoodford — APPROVED** [review]
Left one comment re: circleci config, but ever with that unchanged, LGTM.

File: .circleci/config.yml (hunk 1 above)
Comment (MothOnMars) [reply .circleci/config.yml:83]: Good catch. We do need that step, but I'll rename it to a generic `Wait for Elasticsearch`.

**Review by lorawoodford — APPROVED** [review]
LGTM

**Pull request merged**